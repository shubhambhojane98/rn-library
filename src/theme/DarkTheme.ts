import type {Theme} from '../utils/types';

let DarkTheme: Theme;

DarkTheme = {
  dark: true,
  colors: {
    primary: '#BB86FC',
    primaryVariant: '#3700B3',
    secondary: '#03DAC6',
    secondaryVariant: '#018786',
    background: '#262833',
    error: '#CF6679',
    textColor: '#FFF',
    shadowColor: '#D7D7D7',
    black: '#000',
    white: '#FFF',
    lightgrey: '#D7D7D7',
  },
};

export default DarkTheme;
