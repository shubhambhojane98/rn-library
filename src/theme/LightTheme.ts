import type {Theme} from '../utils/types';

let LightTheme: Theme;

LightTheme = {
  dark: false,
  colors: {
    primary: '#6200EE',
    primaryVariant: '#3700B3',
    secondary: '#03DAC6',
    secondaryVariant: '#018786',
    background: '#FFF',
    error: '#B00020',
    textColor: '#000',
    shadowColor: '#222',
    black: '#000',
    white: '#FFF',
    lightgrey: '#D7D7D7',
    backgroundLayout: 'rgba(0, 0, 0, 0.8)',
    transparent: 'transparent',
    linkColor: '#0000EE',
    greyOpac50: 'rgba(53, 53, 53, 0.5)',
    greyOpac10: 'rgba(53, 53, 53, 0.1)',
  },
};

export default LightTheme;
