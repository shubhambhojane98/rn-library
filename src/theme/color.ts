import {Appearance} from 'react-native';

interface colors {
  yellow: string;
  black: string;
  red: string;
  violet: string;
  taupe: string;
  beige: string;
  grey: string;
  lightgrey: string;
  blue: string;
  darkGrey: string;
  white: string;
  transportation: string;
  logistic: string;
  environment: string;
  sports: string;
  healthcare: string;
  background: string;
  textColor: string;
  placeholderTextColor: string;
  switchBackgroundColor: string;
  paginationDotColor: string;
  paginationDotBgColor: string;
  blackOpac80: string;
  blackOpac60: string;
  blackOpac40: string;
  blackOpac20: string;
  greyOpac80: string;
  greyOpac60: string;
  greyOpac50: string;
  greyOpac40: string;
  greyOpac20: string;
  greyOpac10: string;
  switchActiveBgColor: string;
  switchInActiveBgColor: string;
  switchKnobColor: string;
}

const primaryColors = {
  yellow: '#FAD000',
  black: '#231F20',
  red: '#A81E23',
  violet: '#533467',
  taupe: '#695958',
  beige: '#C3B59F',
  grey: '#353535',
  white: '#fff',
  lightgrey: '#D7D7D7',
  blue: '#007BC1',
  darkGrey: 'rgba(0, 0, 0, 0.2)',
};

const opacColors = {
  blackOpac80: 'rgba(0, 0, 0, 0.8)',
  blackOpac60: 'rgba(0, 0, 0, 0.6)',
  blackOpac40: 'rgba(0, 0, 0, 0.4)',
  blackOpac20: 'rgba(0, 0, 0, 0.2)',
  greyOpac80: 'rgba(53, 53, 53, 0.8)',
  greyOpac60: 'rgba(53, 53, 53, 0.6)',
  greyOpac50: 'rgba(53, 53, 53, 0.5)',
  greyOpac40: 'rgba(53, 53, 53, 0.4)',
  greyOpac20: 'rgba(53, 53, 53, 0.2)',
  greyOpac10: 'rgba(53, 53, 53, 0.1)',
};

const categoryColors = {
  transportation: '#007BC1',
  logistic: '#EF4065',
  environment: '#009A46',
  sports: '#F26524',
  healthcare: '#00ABBE',
};

let themeColors: colors;

const colorScheme = Appearance.getColorScheme();
if (colorScheme === 'dark') {
  // Use dark color scheme
  themeColors = {
    ...primaryColors,
    ...categoryColors,
    ...opacColors,
    background: primaryColors.black,
    textColor: primaryColors.white,
    placeholderTextColor: '#5D5D5D',
    switchBackgroundColor: primaryColors.white,
    paginationDotColor: primaryColors.white,
    paginationDotBgColor: primaryColors.darkGrey,
    switchActiveBgColor: primaryColors.yellow,
    switchInActiveBgColor: primaryColors.white,
    switchKnobColor: primaryColors.black,
  };
} else {
  // Use light color scheme
  themeColors = {
    ...primaryColors,
    ...categoryColors,
    ...opacColors,
    background: primaryColors.yellow,
    textColor: primaryColors.black,
    placeholderTextColor: '#5D5D5D',
    switchBackgroundColor: primaryColors.lightgrey,
    paginationDotColor: primaryColors.white,
    paginationDotBgColor: primaryColors.darkGrey,
    switchActiveBgColor: primaryColors.yellow,
    switchInActiveBgColor: primaryColors.white,
    switchKnobColor: primaryColors.black,
  };
}

export default themeColors;
