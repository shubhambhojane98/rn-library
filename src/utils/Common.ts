export const defaultScale = 0.2;
