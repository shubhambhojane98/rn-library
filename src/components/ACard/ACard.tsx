import React, {FC} from 'react';
import {View, StyleSheet} from 'react-native';
import {Color} from '../../theme';
import {moderateScale} from 'react-native-size-matters';
import {defaultScale} from '../../utils/Common';

interface Props {
  children: any;
  height: number;
  width: number;
  borderRadius?: number;
  borderWidth?: number;
  shadowColor?: string;
  elevation?: number;
  borderColor?: string;
  shadowOffset?: {width: number; height: number};
  shadowOpacity?: number;
  shadowRadius?: number;
}
const TCard: FC<Props> = ({
  height,
  width,
  borderRadius,
  borderWidth,
  shadowColor,
  borderColor,
  children,
  elevation,
  shadowOffset,
  shadowOpacity,
  shadowRadius,
}) => {
  return (
    <View
      style={{
        ...styles.container,
        height: moderateScale(height, defaultScale),
        width: moderateScale(width, defaultScale),
        borderRadius: moderateScale(borderRadius || 20, defaultScale),
        borderWidth: moderateScale(borderWidth || 0, defaultScale),
        borderColor: borderColor,
        elevation: moderateScale(elevation || 5, defaultScale),
        shadowColor,
        shadowOffset,
        shadowOpacity,
        shadowRadius,
      }}>
      {children}
    </View>
  );
};
const styles = StyleSheet.create({
  container: {
    alignSelf: 'center',
    alignContent: 'center',
    margin: moderateScale(5, defaultScale),
    shadowColor: Color.blackOpac80,
    shadowOffset: {
      width: -2,
      height: 4,
    },
    shadowOpacity: 0.2,
    shadowRadius: 3,
  },
});

export default TCard;
