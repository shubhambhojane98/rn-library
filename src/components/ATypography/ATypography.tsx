import React from 'react';
import {StyleProp, StyleSheet, Text, TextProps, TextStyle} from 'react-native';
import {moderateScale} from 'react-native-size-matters';
import {fontFamilies} from '../../constants/fontFamilies';
import {TextAlignment, TypographyVariant} from './ATypographyEnum';
import {defaultScale} from '../../utils/Common';
import {withTheme, useTheme} from '../../core/theming';
import type {Theme} from '../../utils/types';
interface OwnProps {
  variant?: TypographyVariant;
  color?: string;
  style?: StyleProp<TextStyle>;
  textAlign?: TextAlignment;
  fontSize?: number;
  theme: Theme;
}

const defaultFontSize = moderateScale(14, defaultScale);

type Props = OwnProps & TextProps;
const Typography: React.FC<Props> = ({
  variant = TypographyVariant.PRIMARY,
  color,
  style,
  textAlign = TextAlignment.Left,
  children,
  fontSize = defaultFontSize,
  ...otherProps
}) => {
  const {colors} = useTheme();
  return (
    <Text
      style={[
        styles({
          fontSize,
        })[variant],
        {color: color ? color : colors.textColor},
        {textAlign},
        style,
      ]}
      {...otherProps}>
      {children}
    </Text>
  );
};

export default withTheme(Typography);

const styles = (props: {fontSize: number}) =>
  StyleSheet.create({
    primary: {
      fontFamily: `${fontFamilies.Fonts.primary}-Regular`,
      fontSize: moderateScale(props.fontSize, defaultScale),
      fontWeight: 'normal',
    },
    primarySemiBold: {
      fontFamily: `${fontFamilies.Fonts.primary}-SemiBold`,
      fontSize: moderateScale(props.fontSize, defaultScale),
      fontWeight: '600',
    },
    primaryBold: {
      fontFamily: `${fontFamilies.Fonts.primary}-Bold`,
      fontSize: moderateScale(props.fontSize, defaultScale),
      fontWeight: 'bold',
    },
    secondary: {
      fontFamily: `${fontFamilies.Fonts.secondary}-Regular`,
      fontSize: moderateScale(props.fontSize, defaultScale),
      fontWeight: 'normal',
    },
    secondaryDemi: {
      fontFamily: `${fontFamilies.Fonts.secondary}-Demi`,
      fontSize: moderateScale(props.fontSize, defaultScale),
      fontWeight: '600',
    },
    secondaryBold: {
      fontFamily: `${fontFamilies.Fonts.secondary}-Bold`,
      fontSize: moderateScale(props.fontSize, defaultScale),
      fontWeight: 'bold',
    },
  });
