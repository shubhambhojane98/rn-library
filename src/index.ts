import {defaultScale} from './utils/Common';
import {consoleLog, consoleWarn, consoleInfo} from './utils/Log';
import {CheckboxStatus} from './components/ACheckbox/ACheckboxEnum';
import {Color} from './theme';
export {default as AButton} from './components/AButton/AButton';
export {default as ATypography} from './components/ATypography/ATypography';
export {default as AAvatar} from './components/AAvatar/AAvatar';
export {default as ABadge} from './components/ABadge/ABadge';
export {default as AImage} from './components/AImage/AImage';
export {default as ARadioButton} from './components/ARadioButton/ARadioButton';
export {default as ARadioButtonGroup} from './components/ARadioButtonGroup/ARadioButtonGroup';
export {default as ATextInput} from './components/ATextInput/ATextInput';
export {default as ASwitch} from './components/ASwitch/ASwitch';
export {default as AVirtualizedList} from './components/AVirtualizedList/AVirtualizedList';
export {default as AModal} from './components/AModal/AModal';
export {default as ASwitchSelector} from './components/ASwitchSelector/ASwitchSelector';
export {default as AStarRating} from './components/AStarRating/AStarRating';
export {default as ACheckbox} from './components/ACheckbox/ACheckbox';
export {default as AToast} from './components/AToast/AToast';
export {default as ACard} from './components/ACard/ACard';
export {consoleLog, consoleWarn, consoleInfo};
export {Color};
export {CheckboxStatus};
