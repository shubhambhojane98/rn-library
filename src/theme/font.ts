import { Platform, PlatformOSType } from 'react-native';
import { Fonts } from '../utils/types';


export default function configureFonts(config?: {
    [platform in PlatformOSType | 'default']?: Fonts;
  }): Fonts {
    const fonts = Platform.select({ ...config }) as Fonts;
    return fonts;
  }