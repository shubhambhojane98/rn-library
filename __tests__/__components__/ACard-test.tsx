import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ACard from '../../src/components/ACard/ACard';
import ATypography from '../../src/components/ATypography/ATypography';
import {TypographyVariant} from '../../src/components/ATypography/ATypographyEnum';
import {ImageBackground, View} from 'react-native';
import {moderateScale} from 'react-native-size-matters';
import {defaultScale} from '../../src/utils/Common';
import {Color} from '../../src/theme';

it('renders correctly with default prop', () => {
  const tree = renderer
    .create(
      <ACard
        children={
          <ImageBackground
            resizeMode="cover"
            borderRadius={moderateScale(20, defaultScale)}
            style={{flex: 1}}
            source={{
              uri: 'https://blog.logrocket.com/wp-content/uploads/2020/05/setting-up-continuous-deployment-with-react-firebase-github-actions.png',
            }}></ImageBackground>
        }
        height={100}
        width={250}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with height, width, borderRadius, borderWidth, shadowColor,borderColor, children, elevation, shadowOffset, shadowOpacity,shadowRadius,', () => {
  const tree = renderer
    .create(
      <ACard
        width={300}
        height={120}
        borderColor={Color.environment}
        borderWidth={1}
        borderRadius={20}
        elevation={10}
        children={
          <View
            style={{
              flex: 1,
              alignItems: 'center',
              justifyContent: 'center',
              borderRadius: moderateScale(20, defaultScale),
              backgroundColor: Color.environment,
              shadowColor: 'blue',
              elevation: moderateScale(10, defaultScale),
            }}>
            <ATypography variant={TypographyVariant.PRIMARY_BOLD}>
              Title 1
            </ATypography>
            <ATypography variant={TypographyVariant.PRIMARY_SEMI_BOLD}>
              Title 2
            </ATypography>
            <ATypography variant={TypographyVariant.SECONDARY_DEMI}>
              description
            </ATypography>
          </View>
        }
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
