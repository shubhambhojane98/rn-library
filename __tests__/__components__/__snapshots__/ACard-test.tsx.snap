// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with default prop 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "alignSelf": "center",
      "borderColor": undefined,
      "borderRadius": 20,
      "borderWidth": 0,
      "elevation": 5,
      "height": 122.85714285714286,
      "margin": 6.142857142857142,
      "shadowColor": undefined,
      "shadowOffset": undefined,
      "shadowOpacity": undefined,
      "shadowRadius": undefined,
      "width": 307.1428571428571,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Image
      borderRadius={24.57142857142857}
      resizeMode="cover"
      source={
        Object {
          "uri": "https://blog.logrocket.com/wp-content/uploads/2020/05/setting-up-continuous-deployment-with-react-firebase-github-actions.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`renders correctly with height, width, borderRadius, borderWidth, shadowColor,borderColor, children, elevation, shadowOffset, shadowOpacity,shadowRadius, 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "alignSelf": "center",
      "borderColor": "#6200EE",
      "borderRadius": 20,
      "borderWidth": 1,
      "elevation": 10,
      "height": 147.42857142857142,
      "margin": 6.142857142857142,
      "shadowColor": undefined,
      "shadowOffset": undefined,
      "shadowOpacity": undefined,
      "shadowRadius": undefined,
      "width": 368.57142857142856,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderRadius": 24.57142857142857,
        "elevation": 12.285714285714285,
        "flex": 1,
        "justifyContent": "center",
        "shadowColor": "blue",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Mulish-Bold",
            "fontSize": 21.13142857142857,
            "fontWeight": "bold",
          },
          Object {
            "color": "#000",
          },
          Object {
            "textAlign": "left",
          },
          undefined,
        ]
      }
      theme={
        Object {
          "colors": Object {
            "background": "#FFF",
            "backgroundLayout": "rgba(0, 0, 0, 0.8)",
            "black": "#000",
            "error": "#B00020",
            "lightgrey": "#D7D7D7",
            "linkColor": "#0000EE",
            "primary": "#6200EE",
            "primaryVariant": "#3700B3",
            "secondary": "#03DAC6",
            "secondaryVariant": "#018786",
            "shadowColor": "#FFF",
            "textColor": "#000",
            "transparent": "transparent",
            "white": "#FFF",
          },
          "dark": false,
        }
      }
    >
      Title 1
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Mulish-SemiBold",
            "fontSize": 21.13142857142857,
            "fontWeight": "600",
          },
          Object {
            "color": "#000",
          },
          Object {
            "textAlign": "left",
          },
          undefined,
        ]
      }
      theme={
        Object {
          "colors": Object {
            "background": "#FFF",
            "backgroundLayout": "rgba(0, 0, 0, 0.8)",
            "black": "#000",
            "error": "#B00020",
            "lightgrey": "#D7D7D7",
            "linkColor": "#0000EE",
            "primary": "#6200EE",
            "primaryVariant": "#3700B3",
            "secondary": "#03DAC6",
            "secondaryVariant": "#018786",
            "shadowColor": "#FFF",
            "textColor": "#000",
            "transparent": "transparent",
            "white": "#FFF",
          },
          "dark": false,
        }
      }
    >
      Title 2
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "AvenirNextLTPro-Demi",
            "fontSize": 21.13142857142857,
            "fontWeight": "600",
          },
          Object {
            "color": "#000",
          },
          Object {
            "textAlign": "left",
          },
          undefined,
        ]
      }
      theme={
        Object {
          "colors": Object {
            "background": "#FFF",
            "backgroundLayout": "rgba(0, 0, 0, 0.8)",
            "black": "#000",
            "error": "#B00020",
            "lightgrey": "#D7D7D7",
            "linkColor": "#0000EE",
            "primary": "#6200EE",
            "primaryVariant": "#3700B3",
            "secondary": "#03DAC6",
            "secondaryVariant": "#018786",
            "shadowColor": "#FFF",
            "textColor": "#000",
            "transparent": "transparent",
            "white": "#FFF",
          },
          "dark": false,
        }
      }
    >
      description
    </Text>
  </View>
</View>
`;
